### ISSUE TITLE
Discuss extending Neo4J Unit Tests

### ISSUE THREAD

**Issue by noheton** [noheton/shepard#273]
***In GitLab by @xit-max-hey on Oct 21, 2024, 14:03 GMT+2:***

# Refactoring Task

## :mount_fuji: Summary

Currently, most of the Neo4J unit tests, are testing the Neo4j query itself and compare the expected query with the received query. However, none of these queries are actually interpreted by the Neo4j driver. For debugging and improving queries, we should discuss and implement tests that are run on a containerized Neo4j database. Especially, to test built-in driver commands like `findAll()`, or changes in the cypher syntax due to newer Neo4j versions.

## :bulb: Hints

- For this we could use the Quarkus Dev services, after implementing Neo4j in the Quarkus way. [link](https://docs.quarkiverse.io/quarkus-neo4j/dev/index.html#dev-services)
- We could also test things with QuarkusTests against a running neo4j
- Some hints for testing: https://neo4j.com/blog/cypher-and-gql/dark-side-neo4j-worst-practices/

## :thinking: Risks & Open Questions

- We should discuss, if we really want to implement these kind of tests as UnitTests or if it is enough that cypher query interpretation is theoretically handled by the integration tests

*Migrated from GitLab: https://gitlab.com/dlr-shepard/shepard/-/work_items/273*

**Comment by noheton** [comment]
***In GitLab by @xit-max-hey on Oct 22, 2024, 13:26 GMT+2:***

marked this issue as blocked by #274

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Feb 21, 2025, 10:06 GMT+1:***

marked this issue as a duplicate of #492

**Comment by noheton** [comment]
***In GitLab by @xit-paul-reinhardt on Feb 21, 2025, 10:06 GMT+1:***

marked this issue as related to #492
